vumi/components/tests/test_message_store.py

```diff
@@ -112,6 +112,27 @@ def create_outbound_messages(self, batch_id, count, start_timestamp=None,
             messages.append(msg)
         returnValue(messages)
 
+    def _create_event(self, event_type, timestamp):
+        assert event_type in ('ack', 'nack', 'delivery_report')
+        maker = getattr(self.msg_helper, 'make_%s' % (event_type,))
+        return maker(timestamp=timestamp)
```

Comment:
I did try that but we end up needing to duplicate all of the default logic
from the individual `make_ack`, `make_nack` & `make_delivery_report`
methods.